CMake/finders/Findsodium.cmake

```diff
@@ -251,47 +251,49 @@ if (WIN32)
     mark_as_advanced(sodium_DLL_RELEASE)
 endif()
 
-# create imported target
-if(sodium_USE_STATIC_LIBS)
-    set(_LIB_TYPE STATIC)
-else()
-    set(_LIB_TYPE SHARED)
-endif()
-add_library(sodium ${_LIB_TYPE} IMPORTED)
-
-set_target_properties(sodium PROPERTIES
-    INTERFACE_INCLUDE_DIRECTORIES "${sodium_INCLUDE_DIR}"
-    IMPORTED_LINK_INTERFACE_LANGUAGES "C"
-)
+# create imported target only when found, to avoid clashing with in-tree build

```

Comment:
Move leaveinfo_t into a header in libdevilutionx instead of modifying the build system.